src/gsy_e/models/strategy/external_strategies/__init__.py

```diff
@@ -521,16 +521,18 @@ def _publish_trade_event(self, trade, is_bid_trade) -> None:
                                    "local_market_fee":
                                        self.area.current_market.fee_class.grid_fee_rate
                                        if self.area.current_market is not None else "None",
-                                   "attributes": trade.offer_bid.attributes,
-                                   "seller": trade.seller
-                                   if trade.seller_id == self.device.uuid else "anonymous",
-                                   "buyer": trade.buyer
-                                   if trade.buyer_id == self.device.uuid else "anonymous",
-                                   "seller_origin": trade.seller_origin,
-                                   "buyer_origin": trade.buyer_origin,
-                                   "bid_id": trade.offer_bid.id
+                                   "attributes": (trade.match_details["offer"].attributes
+                                                  if trade.is_offer_trade
+                                                  else trade.match_details["bid"].attributes),
+                                   "seller": trade.seller.name
+                                   if trade.seller.uuid == self.device.uuid else "anonymous",
+                                   "buyer": trade.buyer.name
+                                   if trade.buyer.uuid == self.device.uuid else "anonymous",
+                                   "seller_origin": trade.seller.origin,
+                                   "buyer_origin": trade.buyer.origin,
+                                   "bid_id": trade.match_details["bid"].id
                                    if trade.is_bid_trade else "None",
-                                   "offer_id": trade.offer_bid.id
```

Comment:
if both are None, then they will have no id attribute indeed. However this will never be the case, since a Trade object is created either by a bid, or by an offer, or by both.